framework/sandstone.cpp

```diff
@@ -326,34 +326,39 @@ static int test_result_to_exit_code(TestResult result)
 static ChildExitStatus test_result_from_exit_code(forkfd_info info)
 {
     ChildExitStatus r = {};
-    r.extra = info.status;
     if (info.code == CLD_EXITED) {
         // normal exit
         int8_t status = info.status;    // the cast to int8_t transforms 255 into -1
         switch (status) {
         case -1:
             r.result = TestResult::Skipped;
-            r.extra = 0;
             break;
         case EXIT_SUCCESS:
             r.result = TestResult::Passed;
-            r.extra = 0;
             break;
         case EXIT_FAILURE:
             r.result = TestResult::Failed;
-            r.extra = 0;
             break;
         default:
             // a FW problem getting started
             r.result = TestResult::OperatingSystemError;
+            r.extra = info.status;
             break;
         }
-    } else if (info.code == CLD_KILLED || info.code == CLD_DUMPED) {
-        r.result = info.code == CLD_KILLED ? TestResult::Killed : TestResult::CoreDumped;
-        if (info.status == SIGKILL)
+    } else if (info.code == CLD_KILLED) {
+        if (info.status == SIGINT) {
+            r.result = TestResult::Interrupted;
```

Comment:
I'm not sure.

This causes the child having received SIGINT for any reason to declare the testing interrupted, while the current code only considers a test interruption if the *parent* process receives the SIGINT or any of the other interruption signals for that matter. No test in our roster (whether open source or not) currently raises SIGINT or SIGTERM for its workload, but it could happen. We do have tests that use the crashing signals (SIGILL, FPE, SEGV and BUS), after all.

That leaves an external source. SIGHUP and SIGINT sent by the terminal are only sent to the controlling process in the session and that's the parent process.

So I'm leaning towards treating an unexpected SIGINT or SIGHUP as a test crash instead of test interruption.